addons/binding/org.openhab.binding.pidcontroller/src/main/java/org/openhab/binding/pidcontroller/handler/PIDControllerHandler.java

```diff
@@ -0,0 +1,338 @@
+                    logger.debug("Ki adjust {} for channel {}", PIDsetpoint, channelUID);
+                } else {
+                    logger.debug("Channel value can not be null");
+                }
+                break;
+            case kdadjuster:
+                if (channelstate != null) {
+                    Kdadjuster = BigDecimal.valueOf(Double.parseDouble(channelstate.toString()));
+                    logger.debug("valoare de Kd adjust {} for channel {}", PIDsetpoint, channelUID);
+                } else {
+                    logger.debug("Channel value can not be null");
+                }
+                break;
+            case pidlowerlimit:
+                if (channelstate != null) {
+                    PIDOutputLowerLimit = BigDecimal.valueOf(Double.parseDouble(channelstate.toString()));
+                    logger.debug("valoare de PIDLowerLimit adjust {} for channel {}", PIDOutputLowerLimit, channelUID);
+                } else {
+                    logger.debug("Channel value can not be null");
+                }
+                break;
+            case pidupperlimit:
+                if (channelstate != null) {
+                    PIDOutputUpperLimit = BigDecimal.valueOf(Double.parseDouble(channelstate.toString()));
+                    logger.debug("valoare de PIDLowerLimit adjust {} for channel {}", PIDOutputUpperLimit, channelUID);
+                } else {
+                    logger.debug("Channel value can not be null");
+                }
+                break;
+
+        }
+
+    }
+
+    /*
+     * for alpha version this is the call for the PID calculations and posting of the output to the channel output
+     */
+    public void postToOutput() {
+        if (PIDOutputLowerLimit.compareTo(BigDecimal.valueOf(0)) == 0
+                && PIDOutputUpperLimit.compareTo(BigDecimal.valueOf(0)) == 0) {
+            PIDOutputLowerLimit = BigDecimal.valueOf(-255);
+            PIDOutputUpperLimit = BigDecimal.valueOf(255);
+        }
+        logger.debug("Upper Limit :{}", PIDOutputUpperLimit);
+
+        PIDOutput = controller.PIDCalculation(PIDinput, PIDsetpoint, PIDLoopTime, PIDOutputLowerLimit,
+                PIDOutputUpperLimit, Kpadjuster, Kiadjuster, Kdadjuster);
+
+        StringBuilder tempstring2 = new StringBuilder();
+        tempstring2.append(PIDOutput);
+        updateState(outputChannelUID.getId(), DecimalType.valueOf(tempstring2.toString()));
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.eclipse.smarthome.core.thing.binding.BaseThingHandler#channelLinked(org.eclipse.smarthome.core.thing.
+     * ChannelUID)
+     * For now only the output channel is retained in a variable accessible to the class
+     */
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        logger.debug("channelLinked: {}", channelUID);
+        if (channelUID.getId().equals(output))
+

```

Comment:
Please reformat this file using the provided code formatter or clean up settings,